modules/tensor_mechanics/src/materials/ComputeAxisymmetricRZIncrementalStrain.C

```diff
@@ -0,0 +1,46 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ComputeAxisymmetricRZIncrementalStrain.h"
+#include "Assembly.h"
+
+template<>
+InputParameters validParams<ComputeAxisymmetricRZIncrementalStrain>()
+{
+  InputParameters params = validParams<Compute2DIncrementalStrain>();
+  params.addClassDescription("Compute a strain increment and rotation increment for finite strains under axisymmetric assumptions.");
+  return params;
+}
+
+ComputeAxisymmetricRZIncrementalStrain::ComputeAxisymmetricRZIncrementalStrain(const InputParameters & parameters) :
+    Compute2DIncrementalStrain(parameters),
+    _disp_old_0(coupledValueOld("displacements", 0))
+{
+}
+
+Real
+ComputeAxisymmetricRZIncrementalStrain::computeDeformGradZZ(unsigned int qp)
+{
+  if (_assembly.coordSystem() != Moose::COORD_RZ)
```

Comment:
Why does this check have to be done at each qp and each timestep? Just put it in the constructor (or initialSetup if that doesn't work)